api/mesh/v1alpha1/dataplane.proto

```diff
@@ -310,33 +310,5 @@ message Dataplane {
   // defined at a Mesh level.
   MetricsBackend metrics = 2;
 
-  message Probes {
```

Comment:
I don't think we can remove it like this. What about universal?

Reply:
Checked the code, this field only serves Kubernetes scenarios. There is another field called `serviceProbe`. Docs at [here](https://kuma.io/docs/2.8.x/policies/service-health-probes/#universal-probes)

Reply:
I'm pretty sure that's not going to be fine from a backward compat pov

Reply:
Hi @lahabana I don't quite understand why removing this field will break backward compatibility?  
Without this field, we can still parse the existing dataplane objects stored in the DB when people upgrade their versions. So I think this is not a break change.

Reply:
There're might be users who rely on this functionality on Universal. Backward compatibility means not only being able to parse the format, but also preserve the functionality. Usually we mark the field as deprecated and remove it after the 2 minor releases

Reply:
Thanks for the explanation of the "Backward compatibility", I agree with your view of "not only being able to parse the format, but also preserve the functionality". I meant that you guys may be confused this field with the other one `serviceProbe` which serves Universal scenarios. The field "probes" here was only serving Kubernetes scenarios.

Reply:
1. It makes the most sense on k8s, but the usage is not limited to k8s. You can use it on Universal if you want to expose probe without mTLS.
2. Even on k8s I believe you can't delete it, think about the moment when old kuma-dp connects to the new kuma-cp

Reply:
OK, I'm sold. I'll get it back.

Reply:
Please see this comment:
https://github.com/kumahq/kuma/pull/10624#issuecomment-2252162823